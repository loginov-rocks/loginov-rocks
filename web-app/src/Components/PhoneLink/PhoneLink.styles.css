.link:before {
  content: '+' attr(data-ghi) attr(data-abc) attr(data-jkl) attr(data-def);
}
