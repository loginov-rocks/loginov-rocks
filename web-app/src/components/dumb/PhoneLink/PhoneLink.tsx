import * as React from 'react';

import styles from './PhoneLink.styles.css';

interface Props {
  first: string;
  fourth: string;
  second: string;
  third: string;
  title: string;
}

export const PhoneLink: React.FC<Props> = ({
  first, fourth, second, third, title,
}) => {
  const handlePhoneClick = (): void => {
    const phone = `+${first}${second}${third}${fourth}`;
    window.location.href = `tel:${phone}`;
  };

  return (
    <a
      className={styles.link}
      data-abc={second}
      data-def={fourth}
      data-ghi={first}
      data-jkl={third}
      href="#phone"
      onClick={handlePhoneClick}
    >
      {title ? ' ' : ''}
      {title}
    </a>
  );
};
