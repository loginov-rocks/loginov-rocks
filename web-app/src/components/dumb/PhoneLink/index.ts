import { PhoneLink } from './PhoneLink';

export { PhoneLink };
