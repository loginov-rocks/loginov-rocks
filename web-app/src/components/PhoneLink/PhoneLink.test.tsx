/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { PhoneLink } from './PhoneLink';

it('matches snapshot', () => {
  const wrapper = shallow(<PhoneLink first="7 (1" fourth="8-90" second="23) 4" third="56-7" title="Title" />);

  expect(wrapper).toMatchSnapshot();
});
