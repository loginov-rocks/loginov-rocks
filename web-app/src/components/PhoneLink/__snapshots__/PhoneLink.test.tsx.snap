// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="link"
  data-abc="23) 4"
  data-def="8-90"
  data-ghi="7 (1"
  data-jkl="56-7"
  href="#phone"
  onClick={[Function]}
>
   
  Title
</a>
`;
